### PR TITLE
Update slice parameter with additional information

### LINKED ISSUE #2280: [DOC] More detail for slices parameter of reindex API
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [x] Other 

**Tell us about your request.** 
Add more detail about slices parameter of reindex API 

Setting slices to auto will let OpenSearch choose the number of slices to use. This setting will use one slice per shard, up to a certain limit(which is 20). If there are multiple sources, it will choose the number of slices based on the index or backing index with the smallest number of shards.

https://opensearch.org/docs/latest/api-reference/document-apis/reindex/#request-body

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### PR DESCRIPTION
**Pull request by ariamarble**
### Description

Updates slice parameter with additional information

### Issues Resolved

fixes #2280 

### Checklist
- [X] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _api-reference/document-apis/reindex.md [hunk 1]
```diff
@@ -66,7 +66,7 @@ password | Password to authenticate with the remote cluster.
 socket_timeout | The wait time for socket reads. Default is 30s.
 connect_timeout | The wait time for remote connection timeouts. Default is 30s.
 size | The number of documents to reindex.
-slice | Whether to manually or automatically slice the reindex operation so it executes in parallel.
+slice | Whether to manually or automatically slice the reindex operation so it executes in parallel. Setting slices to `auto` allows OpenSearch choose the number of slices to use. This will use one slice per shard, up to a maximum of 20. If there are multiple sources, the number of slices used are based on the index or backing index with the smallest number of shards.
```
Comment (vagimeli) [inline _api-reference/document-apis/reindex.md:69]:
```suggestion
slice | Whether to manually or automatically slice the reindex operation so it executes in parallel. Setting slices to `auto` allows OpenSearch to choose the number of slices to use. `auto` uses one slice per shard, up to a maximum of 20. If there are multiple sources, the number of slices used are based on the index or backing index with the smallest number of shards.
```

Comment (vagimeli) [reply _api-reference/document-apis/reindex.md:69]: Please confirm "This" refers to `auto`. Otherwise, please clarify what "This" refers to.

**Review by vagimeli — APPROVED** [review]
@ariamarble Minor edits.

**Review by hdhalter — COMMENTED** [review]
Should the title be "Reindex document"? For the first line "The reindex document API operation..."

And, do we need to have the 'Introduced 1.0' tag? Seems like this should be reserved for only new features?

File: _api-reference/document-apis/reindex.md (hunk 1 above)
Comment (hdhalter) [reply _api-reference/document-apis/reindex.md:69]: Should this be, "Setting _slice_ to "auto" allows..." 

You could combine the sentences so 'auto' is not the first word in the sentence: Setting this field to `auto` allows OpenSearch to control the number of slices to use, which is one slice per shard, up to a maximum of 20.

**Review by hdhalter — APPROVED** [review]
Looks good! Just a couple of small suggestions.

File: _api-reference/document-apis/reindex.md (hunk 1 above)
Comment (ariamarble) [inline _api-reference/document-apis/reindex.md:69]:
```suggestion
slice | Whether to manually or automatically slice the reindex operation so it executes in parallel. Setting this field to `auto` allows OpenSearch to control the number of slices to use, which is 1 slice per shard, up to a maximum of 20. If there are multiple sources, the number of slices used are based on the index or backing index with the smallest number of shards.
```

**Pull request merged**